code/game/machinery/doors/airlock.dm

```diff
@@ -784,20 +785,55 @@
 /obj/machinery/door/airlock/Topic(href, href_list, var/nowindow = 0)
 	// If you add an if(..()) check you must first remove the var/nowindow parameter.
 	// Otherwise it will runtime with this kind of error: null.Topic()
-	if(!nowindow)
-		..()
-	if(!usr.canUseTopic(src) && !IsAdminGhost(usr))
-		return
-	add_fingerprint(usr)
+	to_chat(world, "In href")
+	if(href_list["doorRef"])
+		to_chat(world, "In href for opening the door")
+		if(text2num(href_list["timeSent"])+300 > world.time && href_list["doorRef"] == REF(src))//href security checks
+
+			to_chat(world, "Checks passed, creating NTNet packet")
+
+			GET_COMPONENT_FROM(target_interface, /datum/component/ntnet_interface, src)
+			if(!target_interface)
+				to_chat(world, "No NTNet interface found in [src], aborting")
+				return
 
-	if((in_range(src, usr) && isturf(loc)) && panel_open)
-		usr.set_machine(src)
+			var/datum/netdata/doorOpenPacket = new//send an NTNet packet to the door
+			doorOpenPacket.recipient_ids = list(target_interface.hardware_id)
+			doorOpenPacket.data["data"] = "open"//command to be sent to airlock (tell it to open)
+			doorOpenPacket.data["data_secondary"] = "toggle"
+			var/obj/item/pda/recieverPDA = locate(href_list["recieverRef"]) in get_viewable_pdas()
+			if(!recieverPDA || !usr.canUseTopic(recieverPDA, BE_CLOSE, FALSE, NO_TK) || !istype(recieverPDA, /obj/item/pda))
+				to_chat(world, "Security check failed")
+				return//more security checks
+			if(!recieverPDA.id)
+				to_chat(usr, "<span class='warning'>The PDA screen displays: \"Error: No ID inserted in PDA.\"</span>")
+				return
+			var/obj/item/card/id/slottedID = recieverPDA.id
+			doorOpenPacket.passkey = slottedID.access
+			ntnet_send(doorOpenPacket)
+			to_chat(usr, "<span class='notice'>The PDA screen displays: \"Door open query sent via NTNet.\"</span>")
+			to_chat(world, "Packet sent")
+			var/obj/item/pda/askerPDA = href_list["senderRef"]
+			askerPDA.door_request_cd = 0
```

Comment:
This probably doesn't work since I don't think the var assignment on 816 is correct. Might have to just remove it